src/runtime_src/core/tools/common/XBMain.cpp

```diff
@@ -79,7 +79,12 @@ void  main_(int argc, char** argv,
   // Parse the command line arguments
   po::variables_map vm;
   po::command_line_parser parser(argc, argv);
-  auto subcmd_options = XBU::process_arguments(vm, parser, allOptions, positionalCommand, false);
+  SubCmd::SubCmdOptions subcmd_options;
```

Comment:
This will still not print an xrt-smi validate specific help. This should be handled in SubCmdValidate::handle_errors_and_validate_tests

Reply:
Actually, this should be already handled by default by an earlier change. This is the behavior I see currently :
```
Z:\Repos\XRT-MCDM-FORK\XRT-MCDM\build\WRelease\xilinx\xrt>xrt-smi validate --run=
ERROR: No test given to validate against.: invalid argument

COMMAND: validate

DESCRIPTION: Validates the given device by executing the platform's validate executable.

USAGE: xrt-smi validate [--help] [-d arg] [-f arg] [-o arg] [-r arg] [--pmode arg]

OPTIONS:
  -d, --device       - The Bus:Device.Function (e.g., 0000:d8:00.0) device of interest
  -f, --format       - Report output format. Valid values are:
                         JSON        - Latest JSON schema
                         JSON-2020.2 - JSON 2020.2 schema
  -o, --output       - Direct the output to the given file
  --pmode            - Specify which power mode to run the benchmarks in. Note: Some tests might be
                       unavailable for some modes
  --help             - Help to use this sub-command
  -r, --run          - Run a subset of the test suite. Valid options are:
                         aie-reconfig-overhead    - Run end-to-end array reconfiguration overhead
                                                    through shim DMA
                         all                      - All applicable validate tests will be executed
                                                    (default)
                         cmd-chain-latency        - Run end-to-end latency test using command
                                                    chaining
```

Reply:
This try/catch is still valid ad we need to catch program_options errors:
```
>xrt-smi validate --ver
XRT build version: 2.18.0
Build hash: 0eeeffe7923c154bc8ff4f61f8373ea5f509314c
Build date: 2024-09-25 19:09:02
Git branch: HEAD
PID: 15892
UID: 0
[Thu Oct 10 16:30:19 2024 GMT]
HOST: XSJKRCK01
EXE:
[xrt-smi] ERROR: option '--ver' is ambiguous and matches '--verbose', and '--version'
```
Without it you'll see this